proposals/0023-asynchronous-program-execution-and-broadcast.md

```diff
@@ -0,0 +1,212 @@
+PoH.
+
+If the leader's proposed block is attached to the heaviest fork,
+the validator can start execution of the UserBlocks optimistically.
+Otherwise the validator should only execute UserBlocks on the
+heaviest fork.
+
+### UserBlock format
+
+UserBlock follows the standard solana ledger format, except it
+doesn't contain any PoH ticks.
+
+### UserBlock broadcast
+
+Builders can form blocks at any time. Each scheduled builder can
+transmit shreds over a pre-allocated shred range for the slot.
+
+For example, if the network is configured with 5k shred blocks, and
+2 builders:
+
+Leader: [0-1000]
+UserBlockSlot 0, Builder 0: [1000-2000]
+UserBlockSlot 0, Builder 1: [2000-3000]
+UserBlockSlot 1, Builder 0: [3000-4000]
+UserBlockSlot 1, Builder 1: [4000-5000]
+
+Leader must fit their entire block in shreds 0 to 1000, including
+PoH tick entries, votes and UserBlockEntries.
+
+Builder 0 transmits shreds 1000-2000 and 3000-4000
+Builder 1 transmits shreds 2000-3000 and 4000-5000
+
+UserBlockEntries for UserBlockSlot 1 must be included by the leader
+after the halfway point in the leaders block.
+
+### BankHash
+
+Votes must include the most recent computed BankHash along with the
+slot for the BankHash computation.  This slot is different
+from the tower vote slot.
+
+Each validator that is computing a BankHash will be able to verify
+the BankHash values created by block producers for previous slots
+because each BankHash has a link to the previous one.
+
+During verification, if a validator sees that > 1/3 of the network
+has incorrectly computed a state transition, they should halt
+immediately.
+
+### VoteHash
+
+All validators will execute all the vote transactions on every fork,
+as they do now. After computing all the vote state transitions the
+validator will compute the VoteHash, identical to the BankHash in
+structure, but covers only the vote state transitions.  The VoteHash
+is included in the vote for the slot that the validator is voting
+for.
+
+### Client Confirmations
+
+Clients connected to a trusted RPC can confirm the transaction as
+soon as the RPC has executed teh UserBlock and the block has been
+optimistically confirmed.
+
+Client can also wait for a status code weighted by stake weight,
+as each validator's execution catches up to the slot and includes
+a BankHash confirming the state of the execution.
+
+It may be sufficient to wait for 1/3+ validators executing the state
+transition instead of the full 2/3+, because if 1/3+ are incorrect
+the network will halt anyways.
+
+## Impact
+
+Multiple nodes can operate as Builder on the network concurrently.
+So clients can pick the nearest one, and the bandwidth to schedule
+and prioritize transactions is doubled.  There needs to be a design
+for BundleTransactions that allow the bundler to prioritize a batch
+of transactions for execution by paying a priority fee for all of
+them, and executing the whole batch together.
+
+Priority fees now also imply execution priority.
```

Comment:
Can you elaborate? Why is that the case now?

Reply:
See the UserBlock execution section.

Reply:
yes, everything is converging.  I think the main difference is that solana has turbine and will provision validators to saturate the bandwidth available to them